src/SprykerEco/Zed/CrefoPayApi/Business/Logger/CrefoPayApiLogger.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\CrefoPayApi\Business\Logger;
+
+use Generated\Shared\Transfer\CrefoPayApiRequestTransfer;
+use Generated\Shared\Transfer\CrefoPayApiResponseTransfer;
+use Generated\Shared\Transfer\PaymentCrefoPayApiLogTransfer;
+use Spryker\Zed\Kernel\Persistence\EntityManager\TransactionTrait;
+use SprykerEco\Zed\CrefoPayApi\Persistence\CrefoPayApiEntityManagerInterface;
+
+class CrefoPayApiLogger implements CrefoPayApiLoggerInterface
+{
+    use TransactionTrait;
+
+    protected const GET_REQUEST_METHOD = 'get%sRequest';
+    protected const GET_RESPONSE_METHOD = 'get%sResponse';
+
+    /**
+     * @var \SprykerEco\Zed\CrefoPayApi\Persistence\CrefoPayApiEntityManagerInterface
+     */
+    protected $entityManager;
+
+    /**
+     * @var string
+     */
+    protected $requestType;
```

Comment:
Could you avoid to store this in the property? We should not have any state in our models.